common/djangoapps/student/tests/test_recent_enrollments.py

```diff
@@ -211,3 +212,26 @@ def test_donate_button_honor_with_price(self):
         self.client.login(username=self.student.username, password=self.PASSWORD)
         response = self.client.get(reverse("dashboard"))
         self.assertNotContains(response, "donate-container")
+
+    @ddt.data(True, False)
+    def test_donate_button_with_enabled_site_configuration(self, enable_donation_site_config):
+        # Enable the enrollment success message and donations
+        self._configure_message_timeout(10000)
+
+        # DonationConfiguration has low precedence if 'ENABLE_DONATIONS' is enable in SiteConfiguration
+        DonationConfiguration(enabled=False).save()
+
+        # Create a white-label course mode
+        # (honor mode with no price set)
+        CourseModeFactory.create(mode_slug="honor", course_id=self.course.id, min_price=0)
```

Comment:
No need to reference "white-label" or "honor" here. You should create an "audit" CourseMode.